econml/tests/test_drlearner.py

```diff
@@ -0,0 +1,558 @@
+        assert (lo <= point).all()
+        assert (point <= hi).all()
+        assert (lo < hi).any()  # for at least some of the examples, the CI should have nonzero width
+
+        interval = dml.const_marginal_effect_interval(np.ones((9, 1)), alpha=0.05)
+        point = dml.const_marginal_effect(np.ones((9, 1)))
+        assert len(interval) == 2
+        lo, hi = interval
+        assert lo.shape == hi.shape == point.shape
+        assert (lo <= point).all()
+        assert (point <= hi).all()
+        assert (lo < hi).any()  # for at least some of the examples, the CI should have nonzero width
+
+        interval = dml.coef__interval(T=2, alpha=0.05)
+        point = dml.coef_(T=2)
+        assert len(interval) == 2
+        lo, hi = interval
+        assert lo.shape == hi.shape == point.shape
+        assert (lo <= point).all()
+        assert (point <= hi).all()
+        assert (lo < hi).any()  # for at least some of the examples, the CI should have nonzero width
+
+    def test_drlearner_all_attributes(self):
+        from sklearn.ensemble import GradientBoostingClassifier, GradientBoostingRegressor, RandomForestRegressor
+        from sklearn.linear_model import LinearRegression, LogisticRegression
+        from econml.utilities import StatsModelsLinearRegression
+        import scipy.special
+        np.random.seed(123)
+        controls = np.random.uniform(-1, 1, size=(5000, 3))
+        T = np.random.binomial(2, scipy.special.expit(controls[:, 0]))
+        sigma = 0.01
+        y = (1 + .5 * controls[:, 0]) * T + controls[:, 0] + np.random.normal(0, sigma, size=(5000,))
+        for X in [controls]:
+            for W in [None, controls]:
+                for sample_weight in [None, 1 + np.random.randint(10, size=X.shape[0])]:
+                    for sample_var in [None, 1 + np.random.randint(10, size=X.shape[0])]:
+                        for featurizer in [None, PolynomialFeatures(degree=2, include_bias=False)]:
+                            for models in [(GradientBoostingClassifier(), GradientBoostingRegressor(),
+                                            RandomForestRegressor(n_estimators=100, max_depth=5, min_samples_leaf=50)),
+                                           (GradientBoostingClassifier(), GradientBoostingRegressor(),
+                                            RandomForestRegressor(n_estimators=100, max_depth=5, min_samples_leaf=50)),
+                                           (LogisticRegression(solver='lbfgs', multi_class='auto'),
+                                            LinearRegression(), StatsModelsLinearRegression())]:
+                                for multitask_model_final in [False, True]:
+                                    if (not isinstance(models, StatsModelsLinearRegression)) and (sample_var is not None):
+                                        continue
+                                    print(X, W, sample_weight, sample_var, featurizer, models, multitask_model_final)

```

Comment:
Any benefit to printing?

Reply:
So that when it crushes I know for which combination of variables it crushed.

Reply:
You can use `self.subTest` for that.  See the DML test for the CATE API.